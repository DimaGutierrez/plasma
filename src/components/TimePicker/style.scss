// TODO(grozki): Disabling this rule because it breaks with either solution tried. Should be attempted again later,
//  maybe a better solution exists.
// stylelint-disable no-descending-specificity
@import '../../styles/base';
@import '../../styles/colors';

// These are here just to shut ESLint up.
/* stylelint-disable block-no-empty */
.borderless {
}

.selectInput {
}

.selectInputError {
}
/* stylelint-enable block-no-empty */

.selectClockIcon {
  background: url('data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIyMCIgaGVpZ2h0PSIyMCIgdmlld0JveD0iMCAwIDIwIDIwIj48cGF0aCBmaWxsPSIjQzFDMUMyIiBmaWxsLXJ1bGU9ImV2ZW5vZGQiIHN0cm9rZT0iI0MxQzFDMiIgc3Ryb2tlLXdpZHRoPSIuNSIgZD0iTTEwIDIuMTI1YTcuODc1IDcuODc1IDAgMSAwIDAgMTUuNzUgNy44NzUgNy44NzUgMCAwIDAgMC0xNS43NXpNMSAxMGE5IDkgMCAxIDEgMTggMCA5IDkgMCAwIDEtMTggMHptOSAwaDQuODc1djEuMTI0aC02di02SDEwVjEweiIvPjwvc3ZnPg==')
  transparent no-repeat center center;
  display: inline-block;
  width: 20px;
  height: 20px;
  padding: 0.5rem;
}

:global {
  :local(.selectInput) {
    width: 7.5rem;
    font-size: 13px;
    color: $cBlack50;

    .time-picker__single-value {
      color: currentColor;
    }

    .time-picker__placeholder {
      color: $cGray400;
    }

    .time-picker__indicators {
      order: 0;
    }

    .time-picker__value-container {
      order: 1;
      padding: 0;
    }

    .time-picker__control {
      border: 1px solid $cGray300;
      border-radius: 2px;
      box-sizing: border-box;
      display: flex;
      flex-grow: 1;
      flex-direction: row;
      font-size: 13px;
      height: 40px;
      justify-content: flex-start;
      outline: none;
      padding: 0;
      text-rendering: geometricPrecision;
      width: 100%;

      &:hover {
        border-color: $cGray200;
      }
    }

    .time-picker__control.time-picker__control--is-focused {
      border-color: $cGray200;
      box-shadow: 0 0 4px 1px $cGray300;
    }

    .time-picker__control.time-picker__control--is-disabled {
      background: white;
      border-color: $cGray400;
      color: $cGray400;
      cursor: not-allowed;

      &:hover {
        border-color: $cGray400;
      }

      .time-picker__placeholder {
        color: currentColor;
      }

      :local(.selectClockIcon) {
        opacity: 0.5;
      }
    }

    .time-picker__option {
      text-align: center;
      color: $cBlack50;

      &:active {
        background-color: $cGray400;
      }

      &--is-focused {
        background-color: $cGray600;
      }

      &--is-selected {
        background-color: $cGray600;
        color: currentColor;
      }
    }

    .time-picker__menu {
      margin-top: 0;
      border-radius: 0 0 2px 2px;
      box-shadow: 0 2px 4px rgba(0, 0, 0, 0.15);
    }
  }

  :local(.selectInput.selectInputError) {
    .time-picker__control {
      border-color: $cWarningRed;

      &:hover {
        border-color: $cWarningRed;
      }
    }

    .time-picker__control--is-focused {
      border-color: $cWarningRed;
    }
  }

  :local(.selectInput.borderless) {
    border-color: transparent;
    box-shadow: none;

    .time-picker__control {
      border-color: transparent;

      &:hover {
        border-color: $cGray200;
        box-shadow: none;
      }

      &--is-focused {
        border-color: $cGray200;
      }
    }
  }
}
// stylelint-enable no-descending-specificity
