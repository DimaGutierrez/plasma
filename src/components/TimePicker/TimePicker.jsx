import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { getDataProps } from '../../dataUtils';
import style from './style.scss';

class TimePicker extends Component {
  static displayName = 'Plasma@TimePicker'

  static propTypes = {
    onBlur: PropTypes.func.isRequired,
    onChange: PropTypes.func.isRequired,
    time: PropTypes.string.isRequired,
    ...getDataProps(),
  }

  handleBlur = (event) => {
    const newTime = event.target.value;

    this.props.onBlur(newTime);
  }

  handleChange = (event) => {
    const newTime = event.target.value;

    this.props.onChange(newTime);
  }

  render() {
    const { time } = this.props;

    return (
      <input
        className={style.component}
        type="time"
        name="timePicker"
        onBlur={this.handleBlur}
        onChange={this.handleChange}
        value={time}
      />
    );
  }
}

export default TimePicker;
