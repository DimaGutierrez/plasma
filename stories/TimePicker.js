import React from 'react';
import { storiesOf } from '@storybook/react';
import TimePicker, { DefaultOptions, FormatTypes } from '../src/components/TimePicker/TimePicker';
import StoryState from '../StoryState';


storiesOf('TimePicker', module)
  .add('default', () => (
    <StoryState
      render={(state, setState) => (
        <TimePicker value={state.value} onChange={value => setState({ value })}/>
      )}
    />))
    .add('disabled', () => (
      <StoryState
        render={(state, setState) => (
          <TimePicker value={state.value} onChange={value => setState({ value })} disabled/>
        )}
      />))
      .add('transparentBackground', () => (
        <StoryState
          render={(state, setState) => (
            <TimePicker
              transparentBackground
              value={state.value || '14:00'}
              onChange={value => setState({ value })}
            />)}
        />
      ))
      .add('24H format', () => (
        <StoryState
          render={(state, setState) => (
            <TimePicker timeFormat={FormatTypes.timeFormat24} defaultOption={DefaultOptions.nextInterval}
                        value={state.value} onChange={value => setState({ value })}/>
          )}
        />
      ))
      .add('12H format', () => (
        <StoryState
          render={(state, setState) => (
            <TimePicker timeFormat={FormatTypes.timeFormat12}  value={state.value || '14:00'} onChange={value => setState({ value })}/>
          )}
        />))
      .add('default option minimum', () => (
        <StoryState
        render={(state, setState) => (
          <TimePicker defaultOption={DefaultOptions.minimum} onChange={value => setState({ value })} value={state.value || "06:00"}/>
        )}
      />))
      .add('default option nextInterval', () => (
        <StoryState
        render={(state, setState) => (
          <TimePicker defaultOption={DefaultOptions.nextInterval} onChange={value => setState({ value })} value={state.value}/>
        )}
      />));
