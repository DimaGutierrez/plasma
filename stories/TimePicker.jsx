import React from 'react';
import { storiesOf } from '@storybook/react';
import 'react-select/dist/react-select.css';
import TimePicker, { DefaultOptions, FormatTypes } from '../src/components/TimePicker/TimePicker';
import StoryState from '../StoryState';

storiesOf('TimePicker', module)
  .add('default', () => (
    <StoryState
      render={(state, setState) => (
        <TimePicker value={state.value} onChange={value => setState({ value })} />
      )}
    />
  ))
  .add('disabled', () => (
    <StoryState
      render={(state, setState) => (
        <TimePicker value={state.value} onChange={value => setState({ value })} disabled />
      )}
    />
  ))
  .add('error', () => (
    <StoryState
      render={(state, setState) => (
        <TimePicker
          defaultOption={DefaultOptions.nextInterval}
          onChange={value => setState({ value })}
          value={state.value}
          error
        />
      )}
    />
  ))
  .add('borderless', () => (
    <StoryState
      render={(state, setState) => (
        <TimePicker
          borderless
          value={state.value || '14:00'}
          onChange={value => setState({ value })}
        />
      )}
    />
  ))
  .add('24H format', () => (
    <StoryState
      render={(state, setState) => (
        <TimePicker
          timeFormat={FormatTypes.timeFormat24}
          defaultOption={DefaultOptions.nextInterval}
          value={state.value || '14:00'}
          onChange={value => setState({ value })}
        />
      )}
    />
  ))
  .add('12H format', () => (
    <StoryState
      render={(state, setState) => (
        <TimePicker
          timeFormat={FormatTypes.timeFormat12}
          value={state.value || '14:00'}
          onChange={value => setState({ value })}
        />
      )}
    />
  ))
  .add('default option minimum', () => (
    <StoryState
      render={(state, setState) => (
        <TimePicker
          defaultOption={DefaultOptions.minimum}
          onChange={value => setState({ value })}
          value={state.value}
        />
      )}
    />
  ))
  .add('default option nextInterval', () => (
    <StoryState
      render={(state, setState) => (
        <TimePicker
          defaultOption={DefaultOptions.nextInterval}
          onChange={value => setState({ value })}
          value={state.value}
        />
      )}
    />
  ));
