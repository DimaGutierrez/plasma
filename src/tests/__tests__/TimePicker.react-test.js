/* eslint-disable */

import React from 'react';
import renderer from 'react-test-renderer';
import TimePicker, { DefaultOptions, FormatTypes } from '../src/components/TimePicker/TimePicker';

describe('TimePicker default', () => {
  test('default', () => {
    const component = renderer.create(<TimePicker disabled />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('24H format', () => {
    const component = renderer.create(
      <TimePicker timeFormat={FormatTypes.timeFormat24} value="14:00" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('12H format', () => {
    const component = renderer.create(
      <TimePicker timeFormat={FormatTypes.timeFormat12} value="14:00" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
