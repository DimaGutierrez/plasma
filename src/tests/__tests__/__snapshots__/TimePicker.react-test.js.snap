// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimePicker default 12H format 1`] = `
<div
  className="Select selectInput has-value Select--single"
  style={undefined}
>
  <div
    className="Select-control"
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={undefined}
  >
    <span
      className="Select-multi-value-wrapper"
      id="react-select-4--value"
    >
      <div
        className="Select-value"
        style={undefined}
        title={undefined}
      >
        <span
          aria-selected="true"
          className="Select-value-label"
          id="react-select-4--value-item"
          role="option"
        >
          02:00 PM
        </span>
      </div>
      <div
        aria-activedescendant="react-select-4--value"
        aria-disabled="false"
        aria-expanded={false}
        aria-label={undefined}
        aria-labelledby={undefined}
        aria-owns=""
        className="Select-input"
        onBlur={[Function]}
        onFocus={[Function]}
        role="combobox"
        style={
          Object {
            "border": 0,
            "display": "inline-block",
            "width": 1,
          }
        }
        tabIndex={0}
      />
    </span>
    <span
      className="Select-arrow-zone"
      onMouseDown={[Function]}
    >
      <span
        className="selectClockIcon"
      />
    </span>
  </div>
</div>
`;

exports[`TimePicker default 24H format 1`] = `
<div
  className="Select selectInput has-value Select--single"
  style={undefined}
>
  <div
    className="Select-control"
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={undefined}
  >
    <span
      className="Select-multi-value-wrapper"
      id="react-select-3--value"
    >
      <div
        className="Select-value"
        style={undefined}
        title={undefined}
      >
        <span
          aria-selected="true"
          className="Select-value-label"
          id="react-select-3--value-item"
          role="option"
        >
          14:00
        </span>
      </div>
      <div
        aria-activedescendant="react-select-3--value"
        aria-disabled="false"
        aria-expanded={false}
        aria-label={undefined}
        aria-labelledby={undefined}
        aria-owns=""
        className="Select-input"
        onBlur={[Function]}
        onFocus={[Function]}
        role="combobox"
        style={
          Object {
            "border": 0,
            "display": "inline-block",
            "width": 1,
          }
        }
        tabIndex={0}
      />
    </span>
    <span
      className="Select-arrow-zone"
      onMouseDown={[Function]}
    >
      <span
        className="selectClockIcon"
      />
    </span>
  </div>
</div>
`;

exports[`TimePicker default default 1`] = `
<div
  className="Select selectInput is-disabled Select--single"
  style={undefined}
>
  <div
    className="Select-control"
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={undefined}
  >
    <span
      className="Select-multi-value-wrapper"
      id="react-select-2--value"
    >
      <div
        className="Select-placeholder"
      >
        Select time
      </div>
      <div
        aria-activedescendant="react-select-2--value"
        aria-disabled="true"
        aria-expanded={false}
        aria-label={undefined}
        aria-labelledby={undefined}
        aria-owns=""
        className="Select-input"
        onBlur={[Function]}
        onFocus={[Function]}
        role="combobox"
        style={
          Object {
            "border": 0,
            "display": "inline-block",
            "width": 1,
          }
        }
        tabIndex={0}
      />
    </span>
    <span
      className="Select-arrow-zone"
      onMouseDown={[Function]}
    >
      <span
        className="selectClockIcon"
      />
    </span>
  </div>
</div>
`;
